src/time.cpp

```diff
@@ -34,6 +35,7 @@ namespace g3 {
          //                    ... also ... This is way more buffer space then we need
 
          auto success = std::strftime(buffer, size, c_time_format, tmb);
+         // TODO: incorrect code
```

Comment:
It's not incorrect but perhaps confusing.

So if the library is built in Debug then for 0 == success then
The assert will always fire. When it fires the assert condition is given (std out/err).
I.e the check in the assert statement is there for the coder more than an actual 
"Real check".

If the library is built in Production then an assert statement is *ignored*. But 
C_time_format is returned instead of the buffer.

I.e the logic is correct